### PR TITLE
Added support for multiple frontend instances and a single Skyline server

### PR DESCRIPTION
**Pull request by michaelshin**
- Refactored the frontend so it no longer spawns a Skyline process. Instead, it tries to connect to an existing Skyline server.
- Changed creating/killing a Skyilne process into connecting/disconnecting to Skyline
- Passed in the entry point/project root when the frontend connects to Skyline

### CODE REVIEW

**Review by jimgao1 — APPROVED** [review]
LGTM!

File: skyline-vscode/src/extension.ts [hunk 1]
```diff
@@ -55,73 +42,35 @@ export function activate(context: vscode.ExtensionContext) {
...
 				};
 
 				let on_close = function() {
-					console.log(sess.backendShouldRestart);
-					if (sess.backendShouldRestart) {
-						console.log("Process closed, restarting.");
+					console.log(sess.resetBackendConnection);
+					console.log("Closing connection to backend");
+					if (sess.resetBackendConnection) {
+						console.log("Reconnecting to backend.");
```
Comment (ssaini4) [inline skyline-vscode/src/extension.ts:55]: are these debug statements? Should we remove them?

File: skyline-vscode/src/extension.ts [hunk 2]
```diff
@@ -55,73 +42,35 @@ export function activate(context: vscode.ExtensionContext) {
...
-							});
-						} else if (stderr.includes("An error occured during analysis")) {
-							console.log("Error, reporting to UI!");
-							sess.report_error(stderr);
-						}
-					});
-
-					skylineProcess.on("close", on_close);
+					vscode.window.onDidChangeActiveTextEditor(editor => {
+						console.log("onDidChangeActiveTextEditor");
```
Comment (ssaini4) [inline skyline-vscode/src/extension.ts:67]: Do we need to log this?

File: skyline-vscode/src/skyline_session.ts [hunk 3]
```diff
@@ -104,6 +100,8 @@ export class SkylineSession {
         console.log("Sending initialization request");
         const message = new pb.InitializeRequest();
         message.setProtocolVersion(5);
```
Comment (ssaini4) [inline skyline-vscode/src/skyline_session.ts:102]: 5 seems like a magic number here

File: skyline-vscode/src/skyline_session.ts [hunk 4]
```diff
@@ -115,17 +113,16 @@ export class SkylineSession {
...
-	console.log("restart_profiling", this.startSkyline, this.skylineProcess);
-        this.backendShouldRestart = true;
-        this.skylineProcess?.kill('SIGKILL');
-	// this.startSkyline && this.startSkyline();
+        console.log("restart_profiling", this.startSkyline);
+        this.resetBackendConnection = true;
+        this.disconnect();
     }
-
+
```
Comment (ssaini4) [inline skyline-vscode/src/skyline_session.ts:125]: You can delete this extra space

**Review by ssaini4 — COMMENTED** [review]
Minor comments

File: skyline-vscode/src/extension.ts (hunk 1 above)
Comment (michaelshin) [reply skyline-vscode/src/extension.ts:55]: Just keeping these comments for now. I will remove all calls to console.log once we're done refactoring

File: skyline-vscode/src/skyline_session.ts (hunk 3 above)
Comment (michaelshin) [reply skyline-vscode/src/skyline_session.ts:102]: It's the current version of the protobuf messages. Perhaps we need to save this in a macro (or the Typescript equivalent)

**Pull request merged**